### PR TITLE
updates: correctly lay new commits on top of old

### LINKED ISSUE #117: [Update] changes from %prep not re-applied
```
$ CONTAINER_CMD='dist2src -vvv convert rpms/catch:c8 source-git/catch:c8' make run
The source-git repository and branch exist. Updating existing source-git...
...

$ cd source-git/catch/
$ git log --pretty=oneline --abbrev-commit
03e2f54 (HEAD -> c8, tag: sg-start, imports/c8/catch-2.2.1-2.el8) Add sources defined in the spec file
e304f76 Add spec-file for the distribution
93b5fe3 .packit.yaml
3a2e8ff catch-2.2.1 base
5ca4f72 Prepare for a new update
9841ab2 (origin/c8) Various changes
a7f80b3 catch-2.2.1 base
608e742 Add spec-file for the distribution
9f180dc .packit.yaml
```

The `9841ab2 (origin/c8) Various changes` commit contains change from [spec's %prep](https://git.centos.org/rpms/catch/blob/c8/f/SPECS/catch.spec#_37). It's reverted in `5ca4f72 Prepare for a new update` but never re-applied.

### PR DESCRIPTION
**Pull request by TomasTomecek**
there are multiple fixes here:
* git_repo.has_ref = bu default, remote refs are not part of
  git.repo.branches - one needs to check repo.remote.refs
* we were not correctly committing changes in BUILD/ git repo which are
  created during %prep
* setting BUILD_repo_path as a property
* when target source git repo branch is not checked out but it's
  present, it's an update

Fixes #117

### CODE REVIEW

File: tests/test_update.py [hunk 1]
```diff
@@ -227,3 +227,52 @@ def test_update_existing(tmp_path: Path, package):
     )
     srpm_path = next(source_git_path.glob("*.src.rpm"))
     assert srpm_path.exists()
+
+
+def test_update_catch(tmp_path: Path):
+    """
+    Jirka found an issue that we cannot update several packages
+    which were created by an old version of d2s, one of them is apr
```
Comment (jpopelka) [inline tests/test_update.py:235]:
```suggestion
    which were created by an old version of d2s, one of them is 'catch'
```

**Review by jpopelka — APPROVED** [review]
:+1:

**Comment by TomasTomecek** [comment]
recheck

**Comment by TomasTomecek** [comment]
recheck

**Comment by TomasTomecek** [comment]
```
fatal: unable to access 'https://git.stg.centos.org/source-git/upower.git/': Failed to connect to git.stg.centos.org port 443: Connection timed out
```
@_@

**Comment by TomasTomecek** [comment]
recheck

**Comment by TomasTomecek** [comment]
recheck

**Comment by TomasTomecek** [comment]
This is how I feel like right now \o/

**Pull request merged**